gcloud/logging/metric.py

```diff
@@ -106,11 +106,8 @@ def from_api_repr(cls, resource, client):
 
         :rtype: :class:`gcloud.logging.metric.Metric`
         :returns: Metric parsed from ``resource``.
-        :raises: :class:`ValueError` if ``client`` is not ``None`` and the
-                 project from the resource does not agree with the project
-                 from the client.
         """
-        metric_name = _metric_name_from_path(resource['name'], client.project)
+        metric_name = resource['name']
```

Comment:
You are correct.  The APIs are maddeningly inconsistent in how they generate such entries.